spec/mongoid/criteria/queryable/selector_spec.rb

```diff
@@ -100,6 +122,28 @@
         end
       end
 
+      context "when merging in a symbol $in with an intersecting value" do
+
+        let(:other) do
+          { "field" => { "$in" => [1] } }
+        end
+
+        before do
+          selector["field"] = { :$in => [1] }
```

Comment:
This test mentions intersection, but would pass if the underlying code was performing a union; I would probably make this test more like `$in: [1,2,3] ∩ $in: [2,3,4] --> $in: [2,3]` to be clear that it is intersecting